InnerEye/ML/reports/classification_report.py

```diff
@@ -176,92 +314,138 @@ def get_metric(val_labels_and_predictions: LabelsAndPredictions,
                             metric=ReportedMetrics.AUC_PR)
-        print_header(f"Area under PR Curve: {pr_auc:.4f}", level=4)
+        metrics["Area under PR Curve"] = pr_auc
 
         optimal_threshold = get_metric(val_labels_and_predictions=val_labels_and_predictions,
                                        test_labels_and_predictions=test_labels_and_predictions,
                                        metric=ReportedMetrics.OptimalThreshold)
-
-        print_header(f"Optimal threshold: {optimal_threshold: .4f}", level=4)
+        metrics["Optimal threshold"] = optimal_threshold
 
     accuracy = get_metric(val_labels_and_predictions=val_labels_and_predictions,
                           test_labels_and_predictions=test_labels_and_predictions,
                           metric=ReportedMetrics.Accuracy,
                           optimal_threshold=optimal_threshold)
-    print_header(f"Accuracy at optimal threshold: {accuracy:.4f}", level=4)
+    metrics["Accuracy at optimal threshold"] = accuracy
 
     fpr = get_metric(val_labels_and_predictions=val_labels_and_predictions,
                      test_labels_and_predictions=test_labels_and_predictions,
                      metric=ReportedMetrics.FalsePositiveRate,
                      optimal_threshold=optimal_threshold)
-    print_header(f"Specificity at optimal threshold: {1 - fpr:.4f}", level=4)
+    metrics["Specificity at optimal threshold"] = 1 - fpr
 
     fnr = get_metric(val_labels_and_predictions=val_labels_and_predictions,
                      test_labels_and_predictions=test_labels_and_predictions,
                      metric=ReportedMetrics.FalseNegativeRate,
                      optimal_threshold=optimal_threshold)
-    print_header(f"Sensitivity at optimal threshold: {1 - fnr:.4f}", level=4)
-    print_header("", level=4)
+    metrics["Sensitivity at optimal threshold"] = 1 - fnr
+
+    return metrics
+
+
+def print_metrics(val_labels_and_predictions: LabelsAndPredictions,
+                  test_labels_and_predictions: LabelsAndPredictions,
+                  is_thresholded: bool = False) -> None:
+    """
+    Given LabelsAndPredictions objects for the validation and test sets, print out some metrics.
+    :param val_labels_and_predictions: LabelsAndPredictions object for the val set. This is used to determine the
+    optimal threshold for classification.
+    :param test_labels_and_predictions: LabelsAndPredictions object for the test set. Metrics are calculated for this
+    set.
+    :param is_thresholded: Whether the model outputs are binary (they have been thresholded at some point)
+                           or are floating point numbers.
+    """
+    metrics = get_all_metrics(val_labels_and_predictions, test_labels_and_predictions, is_thresholded)
+    rows = [[description, f"{value:.4f}"] for description, value in metrics.items()]
+    print_table(rows)
 
 
 def print_metrics_for_all_prediction_targets(val_metrics_csv: Path,
```

Comment:
Any way we can test this to make sure the aggregation of metrics is as expected? I didn't see a test

Reply:
you mean the calculations of the metrics from the right CSV subsets, or the "Mean (std)" column?

Reply:
I meant mean(std)

Reply:
done